tests/performance/event_generator/tests/test_logeventgenerator.py

```diff
@@ -0,0 +1,187 @@
+    (1, 5),
+    (10, 50),
+    (100, 1000),
+])
+def test_log_event_count_parametrized(tmp_path: Path, rate: int, operations: int) -> None:
+    """Test if the correct number of log events are generated with different rates and operations.
+
+    Args:
+        tmp_path (LocalPath): Temporary directory path fixture provided by pytest.
+        rate (int): Number of operation per second.
+        operations (int): number of times logs will be generated.
+    """
+    path = tmp_path / "test.log"
+    generator = LogEventGenerator(
+        rate=rate,
+        path=str(path),
+        operations=operations,
+        max_file_size=None,
+        template_path=None
+    )
+    generator.start()
+    with open(path) as file:
+        lines = file.readlines()
+    assert len(lines) == operations, f"Exactly {operations} log entries should be written."
+
+
+def test_zero_operations(tmp_path: Path) -> None:
+    """Test no log file creation when operations are set to zero.
+
+    Args:
+        tmp_path (LocalPath): Temporary directory path fixture provided by pytest.
+    """
+    path = tmp_path / "test.log"
+    generator = LogEventGenerator(
+        rate=1,
+        path=str(path),
+        operations=0,
+        max_file_size=1,
+        template_path=None
+    )
+    generator.start()
+    assert not path.exists(), "Log file should not exist when no operations are specified."
+
+
+def test_invalid_path() -> None:
+    """Test behavior when an invalid path is provided.
+
+    Expectation:
+        Raises an OSError due to invalid path.
+    """
+    with pytest.raises(OSError):
+        LogEventGenerator(
+            rate=1,
+            path="/invalid/path",
+            operations=1,
+            max_file_size=1,
+            template_path=None
+        )
+
+
+def test_high_rate(tmp_path: Path) -> None:
+    """Test the system's response to a very high event generation rate.
+
+    Args:
+        tmp_path (LocalPath): Temporary directory path fixture provided by pytest.
+    """
+    path = tmp_path / "test.log"
+    generator = LogEventGenerator(
+        rate=10000,
+        path=str(path),
+        operations=100,
+        max_file_size=1,
+        template_path=None
+    )
+    generator.start()
+    assert path.exists(), "Log file should still be created with high rate."
+    with open(path) as file:
+        lines = file.readlines()
+    assert len(lines) == 100, "Exactly 100 log entries should be written."
+
+
+def test_log_rotation(tmp_path: Path) -> None:
+    """Test that log rotation occurs when the max file size is exceeded.
+
+    Args:
+        tmp_path (LocalPath): Temporary directory path fixture provided by pytest.
+    """
+    path = tmp_path / "test.log"
+    generator = LogEventGenerator(
+        rate=10,
+        path=str(path),
+        operations=100,
+        max_file_size=1 / 1024,  # 1 KB in MB
+        template_path=None
+    )
+    generator.start()
+    file_size = os.path.getsize(path)
```

Comment:
Is the file size always 0? After the generator starts and ends, are all files going to be deleted?

Reply:
Fixed [here](https://github.com/wazuh/wazuh-qa/pull/5697/commits/f5a193a34dd69bafe1fad3bc1b660bb83d7b26d1).